src/main/java/io/github/pmckeown/dependencytrack/upload/UploadBomMojo.java

```diff
@@ -60,23 +70,66 @@ public UploadBomMojo(UploadBomAction uploadBomAction, MetricsAction metricsActio
 
     @Override
     public void performAction() throws MojoExecutionException, MojoFailureException {
+        logger.info("Update Project Parent : %s", updateParent);
+
         try {
             if (!uploadBomAction.upload(getBomLocation())) {
                 handleFailure("Bom upload failed");
             }
             Project project = projectAction.getProject(projectName, projectVersion);
+
+            UpdateRequest updateReq = new UpdateRequest();
             if (updateProjectInfo) {
-                logger.info("Updating project info");
-                if (!projectAction.updateProjectInfo(project, getBomLocation())) {
-                    logger.info("Failed to update project info");
+                updateReq.withBomLocation(getBomLocation());
+            }
+            if (updateParent) {
+                updateReq.withParent(getProjectParent(parentName, parentVersion));
+            }
+            if (updateProjectInfo || updateParent) {
+                boolean projectUpdated = projectAction.updateProject(project, updateReq);
+                if (!projectUpdated) {
+                    logger.error("Failed to update project info");
+                    throw new DependencyTrackException("Failed to update project info");
                 }
             }
+
             metricsAction.refreshMetrics(project);
         } catch (DependencyTrackException ex) {
             handleFailure("Error occurred during upload", ex);
         }
     }
 
+    private Project getProjectParent(String parentName, String parentVersion)
+            throws DependencyTrackException {
```

Comment:
I've made some tweaks the implementation of setting the parent too.  

During testing, the error messages were not being displayed and due to my original implementation of error handling that you followed, the `mvn -e` was only showing a generic error so nothing useful for the user.

Reply:
Oops, sorry. I didn't spot this in my testing.